### PR TITLE
Paper plots

### PR DESCRIPTION
**Pull request by kmdeck**
## Purpose 
- Updates evaporation to be less dependent on resolution in the vertical (note: results look worse now)
- ameliorates resolution dep by linearly interpolating to the surface (not efficient as written/allocates a lot)
- Adds two tutorials which make plots included in the paper (moved one from experiments/)
- moves get_deltaz to Domains, as the linear_interpolation_to_surface needs it

## Content

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

File: src/integrated/soil_canopy_model.jl [hunk 1]
```diff
@@ -417,7 +417,7 @@ function soil_boundary_fluxes!(
     t,
 ) where {FT}
     bc = soil.boundary_conditions.top
-    soil_conditions = turbulent_fluxes(bc.atmos, soil, Y, p, t)
+    p.soil.turbulent_fluxes .= turbulent_fluxes(bc.atmos, soil, Y, p, t)
```
Comment (kmdeck) [inline src/integrated/soil_canopy_model.jl:420]: bug fix 👀

File: src/standalone/Soil/energy_hydrology.jl [hunk 2]
```diff
@@ -492,17 +491,22 @@ function ClimaLand.surface_resistance(
...
-            Y.soil.θ_i,
-            model.parameters,
-        ),
+    cds = ClimaCore.Fields.coordinate_field(model.domain.space.subsurface)
+    (; ν, θ_r, hydrology_cm) = model.parameters
+    S_l = effective_saturation.(ν, p.soil.θ_l, θ_r)
+    S_l_sfc = p.soil.sfc_scratch
+    ClimaLand.Domains.linear_interpolation_to_surface!(S_l_sfc, S_l, cds.z)
+    θ_l_sfc = @. S_l_sfc * (ν - θ_r) + θ_r
+    θ_i_sfc = ClimaLand.Domains.top_center_to_surface(Y.soil.θ_i)
```
Comment (juliasloan25) [inline src/standalone/Soil/energy_hydrology.jl:500]: Maybe we can change this to use `linear_interpolation_to_surface` now too, since we know that function isn't allocating a new field. Or we could have a separate PR to go through and replace `top_center_to_surface` with `linear_interpolation_to_surface!` throughout the code

File: docs/tutorials/standalone/Soil/freezing_front.jl [hunk 3]
```diff
@@ -106,7 +106,7 @@ hydrology_cm = vanGenuchten{FT}(; α = vg_α, n = vg_n);
 #hcm = BrooksCorey(;ψb = ψb, c = c);
 θ_r = FT(0.05)
 ν_ss_om = FT(0.3)
-ν_ss_quartz = FT(0.7)
+ν_ss_quartz = FT(0.3)
```
Comment (juliasloan25) [inline docs/tutorials/standalone/Soil/freezing_front.jl:109]: Why is this changed? I also noticed it has value FT(1) in evaporation.jl. Should it be unified across files?

File: docs/tutorials/standalone/Soil/evaporation_gilat_loess.jl [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+    0.34 / (1 + exp(-(z + 0.165) / 0.005)) + 0.05
+end
+function init_soil!(Y, z, params)
+    FT = eltype(Y.soil.ϑ_l)
+    Y.soil.ϑ_l .= estimated_ic.(z)
+    Y.soil.θ_i .= 0
+    T = FT(294.15)
+    ρc_s = @. Soil.volumetric_heat_capacity(Y.soil.ϑ_l, FT(0), params)
+    Y.soil.ρe_int =
+        Soil.volumetric_internal_energy.(FT(0), ρc_s, T, Ref(params))
```
Comment (juliasloan25) [inline docs/tutorials/standalone/Soil/evaporation_gilat_loess.jl:164]: Is the `FT(0)` in these two lines `Ksat`? If so, it would be good to set it on another line to clarify

File: docs/tutorials/standalone/Soil/evaporation_gilat_loess.jl [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+    top = top_bc,
+    bottom = WaterHeatBC(; water = zero_water_flux, heat = zero_heat_flux),
+)
+
+# Without drainage - wet boundary at bottom
+bottom_water_bc = MoistureStateBC((p, t) -> 0.35)
+no_drainage_boundary_fluxes = (;
+    top = top_bc,
+    bottom = WaterHeatBC(; water = bottom_water_bc, heat = zero_heat_flux),
+)
```
Comment (juliasloan25) [inline docs/tutorials/standalone/Soil/evaporation_gilat_loess.jl:132]: I think it would be more clear/readable to define each boundary flux in the case it's used for, but this is personal preference :)

**Review by juliasloan25 — APPROVED** [review]
Looks good! I had a few comments, but I think it's good to merge

File: src/standalone/Soil/energy_hydrology.jl (hunk 2 above)
Comment (kmdeck) [reply src/standalone/Soil/energy_hydrology.jl:500]: I will change it here, and then open an issue for the other ones

Comment (kmdeck) [reply src/standalone/Soil/energy_hydrology.jl:500]: oh wait. it will be a different issue. because we are in the same function, but we only have one allocated scratch spot to hold the values (instead of alllocating), we cant reuse it in the same function. I think we need multiple surface scratch fields...

File: docs/tutorials/standalone/Soil/freezing_front.jl (hunk 3 above)
Comment (kmdeck) [reply docs/tutorials/standalone/Soil/freezing_front.jl:109]: I changed it back - I dont think we have measurements for it, so we dont really know what it will be, but I also dont think it makes a big difference. so might as well keep it the same.

In general this parameter is soil specific (clay vs sand, etc) , so it is OK for it to change across files if those files represent different experiments.

File: docs/tutorials/standalone/Soil/evaporation_gilat_loess.jl (hunk 4 above)
Comment (kmdeck) [reply docs/tutorials/standalone/Soil/evaporation_gilat_loess.jl:164]: I defined it in another line, good call!

File: docs/tutorials/standalone/Soil/evaporation_gilat_loess.jl (hunk 5 above)
Comment (kmdeck) [reply docs/tutorials/standalone/Soil/evaporation_gilat_loess.jl:132]: good idea! I changed it

File: src/standalone/Soil/energy_hydrology.jl (hunk 2 above)
Comment (kmdeck) [reply src/standalone/Soil/energy_hydrology.jl:500]: noted here: https://github.com/CliMA/ClimaLand.jl/issues/615
